test/utils/Arrays.test.js

```diff
@@ -94,28 +110,30 @@ contract('Arrays', function () {
         artifact: AddressArraysMock,
         elements: Array(10)
           .fill()
-          .map(() => web3.utils.randomHex(20)),
+          .map(() => randomHex(20)),
       },
       {
         type: 'bytes32',
         artifact: Bytes32ArraysMock,
         elements: Array(10)
           .fill()
-          .map(() => web3.utils.randomHex(32)),
+          .map(() => randomHex(32)),
       },
       {
         type: 'uint256',
         artifact: Uint256ArraysMock,
         elements: Array(10)
           .fill()
-          .map(() => web3.utils.randomHex(32)),
+          .map(() => randomHex(32)),
       },
     ]) {
       it(type, async function () {
-        const contract = await artifact.new(elements);
+        const contract = await ethers.deployContract(artifact, [elements]);
 
         for (const i in elements) {
-          expect(await contract.unsafeAccess(i)).to.be.bignumber.equal(elements[i]);
+          expect(await contract.unsafeAccess(i)).to.be.equal(
+            type == 'address' ? ethers.getAddress(elements[i]) : elements[i],
+          );
```

Comment:
lets cleanup the elements before.
```suggestion
          expect(await contract.unsafeAccess(i)).to.be.equal(elements[i]);
```